kotsadm/web/src/components/apps/AppSnapshotSchedule.jsx

```diff
@@ -118,17 +121,43 @@ class AppSnapshotSchedule extends Component {
     this.setState({ selectedRetentionUnit: retentionUnit });
   }
 
-  componentDidUpdate = (lastProps) => {
-    if (this.props.snapshotConfig.snapshotConfig && this.props.snapshotConfig.snapshotConfig !== lastProps.snapshotConfig.snapshotConfig) {
+  getSnapshotConfig = async () => {
+    this.setState({ loadingConfig: true, gettingConfigErrMsg: "", displayErrorModal: false });
+    await fetch(`${window.env.API_ENDPOINT}/app/${this.props.app.slug}/snapshot/config`, {
+      method: "GET",
+      headers: {
+        "Authorization": Utilities.getToken(),
+        "Content-Type": "application/json",
+      }
+    }).then(async (res) => {
```

Comment:
i would recommend using the `async/await` in a `try/catch` syntax. it gives the choice of `awaiting` to the caller of the function. (this function is called without await, so in this case, you can also just get rid of `async/await` here, i would recommend the first approach though).